modules/light-clients/06-solomachine/types/update.go

```diff
@@ -84,6 +89,31 @@ func (cs ClientState) VerifyClientMessage(cdc codec.BinaryCodec, clientMsg expor
 	return nil
 }
 
+// CheckForMisbehaviour returns true for type Misbehaviour (passed VerifyClientMessage check), otherwise returns false
+func (cs ClientState) CheckForMisbehaviour(
+	_ sdk.Context, _ codec.BinaryCodec, _ sdk.KVStore,
+	msg exported.Header, // TODO: Update to exported.ClientMessage
+) bool {
+	var foundMisbehaviour bool
+	switch msg.(type) {
+	case *Header:
+		foundMisbehaviour = false
+	case *Misbehaviour:
+		foundMisbehaviour = true
+	}
```

Comment:
ma
```suggestion
	switch msg.(type) {
	case *Misbehaviour:
		return true
	default: 
	    return false
	}
```
Will this work? And then you can remove the local var

Reply:
Yeah sure. I was also considering removing the type switch entirely and just using type assertion:

```
if _, ok := msg.(*Misbehaviour); ok {
    return true
}

return false
```

Either works really I guess!

Reply:
That sounds good to me to!